solr/core/src/java/org/apache/solr/storage/TeeDirectory.java

```diff
@@ -450,6 +456,17 @@ public long getFilePointer() {
     public long getChecksum() throws IOException {
       return primary.getChecksum();
     }
+
+    @Override
+    public long getBytesWritten() {
+      if (primary instanceof CompressingDirectory.SizeReportingIndexOutput) {
+        return ((CompressingDirectory.SizeReportingIndexOutput) primary).getBytesWritten();
+      } else if (secondary instanceof CompressingDirectory.SizeReportingIndexOutput) {
+        return ((CompressingDirectory.SizeReportingIndexOutput) secondary).getBytesWritten();
+      } else {
+        return getFilePointer();
+      }
+    }
```

Comment:
This is a little weird for `TeeDirectory`; it's effectively defining the contract of `getBytesWritten()` as "if some delegate directory implements `SizeReportingIndexOutput`, then report its value of `getBytesWritten`, arbitrarily.

I think it would be clearer to always pull `getBytesWritten()` from `secondary` (which will be CompressingDirectory). Still a bit weird because technically `primary` would also have written uncompressed bytes to a disk, but I think we can define the contract such that this is ok.

(Note also that only taking `getBytesWritten()` from `secondary` would make this compatible with/analogous to `TeeDirectory.onDiskFileLength()`, which only ever takes this value from the persistent copy as well.)